### PR TITLE
Refactor ItemLookup to fix two minor suggestion-related bugs

### PR DESCRIPTION
**Pull request by lucaswerkmeister**
See the individual commit messages; I think it makes sense to have this as one pull request, but merge it as ~~two~~ three separate commits (merge with rebase).

The refactoring also prepares for further changes to the `ItemLookup` that will be made in T308118.

### CODE REVIEW

**Comment by micgro42** [comment]
The first commit is fine, but the second one seems to conceptually conflict with #185, right?

Also, as the second commit message mentions, we might get outdated results from the API. I wonder if we should check for that and discard those that were not done for the current input. Codex has an example for this: https://doc.wikimedia.org/codex/main/components/lookup.html#demos (see the code of "with fetched results")

**Comment by lucaswerkmeister** [comment]
> The first commit is fine, but the second one seems to conceptually conflict with #185, right?

I’d see #185 more as a further development of this PR… I guess we could replace the second commit with the resulting tree of #185, but to me it also makes sense on its own. But maybe I can take another look at this tomorrow and see if I change my mind ^^

> Also, as the second commit message mentions, we might get outdated results from the API. I wonder if we should check for that and discard those that were not done for the current input. Codex has an example for this: https://doc.wikimedia.org/codex/main/components/lookup.html#demos (see the code of "with fetched results")

Hm, to me it seems okay to use the outdated results as long as they don’t override the newer results – they could still be useful. So some kind of check would make sense, but I wouldn’t go with the `if ( currentSearchTerm.value === value )`, but rather something that guards only against requests fulfilling in a different order than we sent them. (Track a “generation number” of the search results along with the list of results, or something like that…)

**Comment by lucaswerkmeister** [comment]
> > The first commit is fine, but the second one seems to conceptually conflict with #185, right?
> 
> I’d see #185 more as a further development of this PR… I guess we could replace the second commit with the resulting tree of #185, but to me it also makes sense on its own. But maybe I can take another look at this tomorrow and see if I change my mind ^^

I think I still prefer keeping them separate. First split the list into two but assign it in roughly the same place, then change how one of the lists is assigned / generated. To me it seems more understandable this way.

**Comment by lucaswerkmeister** [comment]
(Rebased without changes, just want to make sure the merge from main isn’t included when this is rebase-merged.)

**Review by micgro42 — APPROVED** [review]

File: src/components/ItemLookup.vue [hunk 1]
```diff
@@ -81,9 +88,9 @@ const onSearchInput = ( inputValue: string ) => {
 const onScroll = async () => {
 	const searchReults = await props.searchForItems(
 		searchInput.value,
-		searchSuggestions.value.length,
+		searchedOptions.value.length,
```
[the remark this answers is not in the document]
Comment (lucaswerkmeister) [reply src/components/ItemLookup.vue:91]: Hm, you’re right, I didn’t realize that.

#185 fixes that, right?

[the remark this answers is not in the document]
Comment (lucaswerkmeister) [reply src/components/ItemLookup.vue:90]: Should be fixed now, I pulled the #185 commit into this PR as well.

File: src/components/ItemLookup.vue [hunk 2]
```diff
@@ -25,61 +25,60 @@ const emit = defineEmits( {
...
-const searchInput = ref( '' );
 const onSearchInput = ( inputValue: string ) => {
 	searchInput.value = inputValue;
 	if ( inputValue.trim() === '' ) {
-		searchSuggestions.value = [];
+		searchedOptions.value = [];
 		return;
 	}
 
-	if ( inputValue === lastSelectedOption.value?.label ) {
```
[the remark this answers is not in the document]
Comment (micgro42) [reply src/components/ItemLookup.vue:82]: yes, I think it is still ok, and it will be aligned properly when we switch to Codex

**Review by micgro42 — APPROVED** [review]

**Pull request merged**